docs/guides/using-mtls-auth-plugin.md

```diff
@@ -0,0 +1,291 @@
+    EF8F+ka1RaAYWTsXI1BAkJbb3kdo/yUCAwEAAaMTMBEwDwYDVR0TBAgwBgEB/wIB
+    ADANBgkqhkiG9w0BAQsFAAOCAQEAVvB/PeVZpeQ7q2IQQQpADtTd8+22Ma3jNZQD
+    EkWGZEQLkRws4EJNCCIvkApzpx1GqRcLLL9lbV+iCSiIdlR5W9HtK07VZ318gpsG
+    aTMNrP9/2XWTBzdHWaeZKmRKB04H4z7V2Dl58D+wxjdqNWsMIHeqqPNKGamk/q8k
+    YFNqNwisRxMhU6qPOpOj5Swl2jLTuVMAeGWBWmPGU2MUoaJb8sc2Vix9KXcyDZIr
+    eidkzkqSrjNzI0yJ2gdCDRS4/Rw9iV3B3SRMs0mJMLBDrsowhNfLAd8I3NHzLwps
+    dZFcvZcT/p717K3hlFVdjGnKIgKcG7aYji/XRR87HKnc+cJMCw==
+    -----END CERTIFICATE-----
+  id: cce8c384-721f-4f58-85dd-50834e3e733a" | kubectl create -f -
+secret/my-ca-cert created
+```
+
+Please note the ID, you can use this ID one or use a different one but
+the ID is important in the next step when we create the plugin.
+Each CA certificate that you create needs a unique ID.
+Any random UUID will suffice here and it doesn't have an security
+implication.
+
+## Configure mtls-auth plugin
+
+Next, we are going to create an `mtls-auth` KongPlugin resource which references
+CA certificate provisioned in the last step:
+
+```bash
+$ echo "
+apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  name: mtls-auth
+config:
+  ca_certificates:
+  - cce8c384-721f-4f58-85dd-50834e3e733a
+  skip_consumer_lookup: true
+  revocation_check_mode: SKIP
+plugin: mtls-auth
+" | kubectl apply -f -
+kongplugin.configuration.konghq.com/mtls-auth created
+```
+
+## Install a dummy service
+
+Let's deploy an echo service which we wish to protect
+using TLS client certificate authentication.
+
+```bash
+$ kubectl apply -f https://bit.ly/echo-service
+service/echo created
+deployment.apps/echo created
+```
+
+You can deploy a different service or skip this step if you already
+have a service deployed in Kubernetes.
+
+## Setup Ingress
+
+Let's expose the echo service outside the Kubernetes cluster
+by defining an Ingress.
+
+```bash
+$ echo "
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: demo
+  annotations:
+    konghq.com/plugins: mtls-auth
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /foo
+        backend:
+          serviceName: echo
+          servicePort: 80
+" | kubectl apply -f -
+ingress.extensions/demo created
+```
+
+## Test the endpoint
+
+Now, let's test to see if Kong is asking for client certificate
+or not when we make the request:
+
+```
+$ curl -k https://$PROXY_IP
+HTTP/2 401
+date: Mon, 11 May 2020 18:15:05 GMT
+content-type: application/json; charset=utf-8
+content-length: 50
+x-kong-response-latency: 0
+server: kong/2.0.4.0-enterprise-k8s
+
+{"message":"No required TLS certificate was sent"}
+```
+
+As we can see, Kong is restricting the request because it doesn't
+have the necessary authentication information.
+
+Two things to note here:
+- `-k` is being used because Kong is setup to serve a self-signed
+  certificate. For real use-cases and making the authentication mutual,
+  you will need to setup Kong to serve a valid certificate that is
+  signed by a trusted CA in your network.
```

Comment:
"set up" (the single word is typically only used as a noun) and "use cases" (most nouns are not hyphenated, see https://docs.microsoft.com/en-us/style-guide/punctuation/dashes-hyphens/hyphens)

Structure of second sentence is a bit awkward and includes some unnecessary qualifiers, suggest:
`-k` is used because Kong is set up to serve a self-signed certificate by default. For full mutual authentication in production use cases, you must configure Kong to serve a certificate that is signed by a trusted CA.